### ISSUE TITLE
Model.find.then() results in MongoInvalidArgumentError: Method "collection.find()" accepts at most two arguments

### ISSUE THREAD

**Issue by nwongsavit** [Automattic/mongoose#10634]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
`Model.find().then()` results in `MongoInvalidArgumentError: Method "collection.find()" accepts at most two arguments`.

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Connect to Mongo DB using Mongoose (6.0.2)
2. Chain `Model.find()` with `.then()`
3.  Get a `MongoInvalidArgumentError` on `collection.find()`

Code:
```Node
Person.find({}).then(result => {
  result.forEach(person => {
    console.log(person)
  })

  mongoose.connection.close()
})
```

Error:
```Bash
(node:7142) UnhandledPromiseRejectionWarning: MongoInvalidArgumentError: Method "collection.find()" accepts at most two arguments
    at Collection.find (/Users/nwongsavit/Projects/fso/part3/phonebook/node_modules/mongodb/lib/collection.js:238:19)
    at NativeCollection.<computed> [as find] (/Users/nwongsavit/Projects/fso/part3/phonebook/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:191:33)
    at NativeCollection.Collection.doQueue (/Users/nwongsavit/Projects/fso/part3/phonebook/node_modules/mongoose/lib/collection.js:135:23)
    at /Users/nwongsavit/Projects/fso/part3/phonebook/node_modules/mongoose/lib/collection.js:82:24
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:7142) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:7142) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**What is the expected behavior?**
Should be able to chain `.then()` to `Model.find()` without an error.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node: v14.15.4
Mongoose: 6.0.2
MongoDB: 4.1.1

**Comment by Yoyomo1** [comment]
I am having the same problem. I found a post that says downgrading to version 5.13.8 gets rid of the issue. [https://stackoverflow.com/questions/68957133/mongoose-model-find-throws-a-unhandledpromiserejectionwarning-error](url)

**Comment by AndrewBarba** [comment]
Same issue. Downgrading to 6.0.1 also works.

**Comment by AndrewBarba** [comment]
Similar to previous issue... `await`ing the default connection before running any queries works correctly

**Comment by lorand-horvath** [comment]
This is a duplicate of the issue I reported here initially https://github.com/Automattic/mongoose/issues/10610#issuecomment-907060604 and should be fixed in 6.0.3 via https://github.com/Automattic/mongoose/commit/9de60a771e37cc2fcb7af3d805294b49239aed17
